### DISCUSSION TITLE
react native + useForm + Controller - errors and touchedFields do not update when using simulators

### DISCUSSION THREAD

**Discussion by arifb** [react-hook-form/react-hook-form#5668] [Q&A] [answered]
Hi - thanks for the great library. Following this example https://snack.expo.io/@bluebill1049/react-hook-form-v7---controller which of course works perfectly on snack. I simply added touchedFields to the example, here's the code: https://snack.expo.io/@arifb/react-hook-form-v7---controller

However when using the same code on an android simulator on a windows box or using the iOS simulator on a mac, touchedFields never gets populated (always an empty object) and on android, errors never gets populated (but strangely in the iOS simulator it does).

So in summary:
on iOS sim and Android sim, touchedFields always an empty object
on Android sim, errors always an empty object

Pretty strange - any ideas why this could be?

Will try on device shortly, hopefully all will be well but thought I should put on your radar.

**Comment by bluebill1049** [comment]
can you try to reproduce this issue?

**Comment by jameskennethrobinson** [comment]
This also seems to happening for me... @arifb did the problem persist when you tried on a real device?

**Comment by tomasmozeris** [comment]
I can reproduce this on iOS simulator and real device, `touchedFields` is always empty object with `7.11.1` on react native

**Reply by bluebill1049** [reply]
did you wire up in `onBlur` from field object?

**Reply by tomasmozeris** [reply]
thanks @bluebill1049, I missed `onBlur` and it works on iOS simulator and real device, thanks again on quick feedback

**Comment by bluebill1049** [comment, marked as answer]
make sure `onBlur` prop is wired up in your component.
